lib/asyn-thread.c

```diff
@@ -454,6 +634,24 @@ static bool init_resolve_thread(struct Curl_easy *data,
   /* The thread will set this to 1 when complete. */
   td->tsd.done = 0;
 
+#if (_WIN32_WINNT >= 0x0600)
+  if(Curl_isWindows8OrGreater) {
+    wsprintfW(td->tsd.w8.name, L"%S", hostname);
```

Comment:
This must first check that the host name fits the target buffer as curl does not generally restrict host name lengths to 255 bytes.

Reply:
I'm not familiar with wsprintfW(), but does this convert the name somehow? Or put another way: is it necessary or could this function be skipped?

Reply:
> I'm not familiar with wsprintfW(), but does this convert the name somehow?

wsprintfW (notice the 'W' suffix for wide char) and %S (notice the uppercase 'S') means write a char string (eg hostname) as part of a wide char string. (The MS documentation is confusing. I tried to clarify it at MicrosoftDocs/sdk-api#1713). It's only safe if the wide character buffer is big enough.

curlx_convert_UTF8_to_wchar could be used instead but since the call to GetAddrInfoExW is asynchronous I would free the wide character copy of hostname only after the completion routine is called.

Reply:
i updated the code use `MultiByteToWideChar` (similar code used by `curlx_convert_UTF8_to_wchar` under the hood)